src/Tasks/Microsoft.NET.Build.Tasks/ResolvePackageAssets.cs

```diff
@@ -1405,6 +1438,203 @@ private void WritePackageDependencies()
+                            .Select(pkg => pkg.Name),
+                        StringComparer.OrdinalIgnoreCase);
+
+                    foreach (var package in _compileTimeTarget.Libraries)
+                    {
+                        if (projectFileDependencies.Contains(package.Name))
+                        {
+                            string packageId = GetPackageId(package);
+
+                            // itemSpec
+                            dependencies.Add(packageId);
+                        }
+                    }
+
+                    return dependencies;
+                }
+
+                static string GetPackageId(LockFileTargetLibrary package) => $"{package.Name}/{package.Version.ToNormalizedString()}";
+
+                // get library and file definitions
+                List<PackageDefinitionItem> GetPackageDefinitions()
+                {
+                    List<PackageDefinitionItem> packageDefinitions = new();
+
+                    foreach (var package in _lockFile.Libraries)
+                    {
+                        PackageDefinitionItem item = new PackageDefinitionItem();
+
+                        var packageName = package.Name;
+                        var packageVersion = package.Version.ToNormalizedString();
+                        string packageId = $"{packageName}/{packageVersion}";
+
+                        item.Name = packageName;
+                        item.Type = package.Type;
+                        item.Version = packageVersion;
+                        item.Path = package.Path ?? string.Empty;
+
+                        string resolvedPackagePath = ResolvePackagePath(package);
+                        item.ResolvedPath = resolvedPackagePath ?? string.Empty;
+
+                        item.DiagnosticLevel = GetPackageDiagnosticLevel(package);
+
+                        packageDefinitions.Add(item);
+                    }
+
+                    return packageDefinitions;
+                }
+
+                string ResolvePackagePath(LockFileLibrary package)
+                {
+                    if (package.IsProject())
+                    {
+                        var relativeMSBuildProjectPath = package.MSBuildProject;
+
+                        if (string.IsNullOrEmpty(relativeMSBuildProjectPath))
+                        {
+                            throw new BuildErrorException(Strings.ProjectAssetsConsumedWithoutMSBuildProjectPath, package.Name, _task.ProjectAssetsFile);
+                        }
+
+                        return GetAbsolutePathFromProjectRelativePath(relativeMSBuildProjectPath);
+                    }
+                    else
+                    {
+                        return _packageResolver.GetPackageDirectory(package.Name, package.Version);
+                    }
+                }
+
+                string GetAbsolutePathFromProjectRelativePath(string path)
```

Comment:
We should also be able to delete this method.

Reply:
done